services/web/server/src/simcore_service_webserver/projects/projects_db.py

```diff
@@ -311,63 +311,51 @@ async def add_project(
-            user_groups: List[RowProxy] = await self.__load_user_groups(conn, user_id)
-            query = textwrap.dedent(
-                f"""\
-                SELECT *
-                FROM projects
-                WHERE projects.type != 'TEMPLATE'
-                AND (jsonb_exists_any(projects.access_rights, array[{', '.join(f"'{group.gid}'" for group in user_groups)}])
-                OR prj_owner = {user_id})
-                """
-            )
-            projects_list = await self.__load_projects(
-                conn, query, user_id, user_groups, filter_by_services=filter_by_services
-            )
-
-        return projects_list
-
-    async def load_template_projects(
-        self,
-        user_id: int,
+        user_id: PositiveInt,
         *,
-        only_published=False,
+        filter_by_project_type: Optional[ProjectType] = None,
         filter_by_services: Optional[List[Dict]] = None,
-    ) -> List[Dict]:
-        log.info("Loading public template projects")
-
-        # TODO: eliminate this and use mock to replace get_user_project instead
-        projects_list = [prj.data for prj in Fake.projects.values() if prj.template]
-
+        only_published: Optional[bool] = False,
+        offset: Optional[int] = 0,
+        limit: Optional[int] = None,
+    ) -> Tuple[List[Dict[str, Any]], List[ProjectType], int]:
+        project_type_where_clause = (
+            f"= '{filter_by_project_type.value}'"
+            if filter_by_project_type
+            else "IS NOT NULL"
+        )
         async with self.engine.acquire() as conn:
             user_groups: List[RowProxy] = await self.__load_user_groups(conn, user_id)
-
-            # NOTE: in order to use specific postgresql function jsonb_exists_any we use raw call here
             query = textwrap.dedent(
                 f"""\
                 SELECT *
                 FROM projects
-                WHERE projects.type = 'TEMPLATE'
+                WHERE projects.type {project_type_where_clause}
                 {'AND projects.published ' if only_published else ''}
                 AND (jsonb_exists_any(projects.access_rights, array[{', '.join(f"'{group.gid}'" for group in user_groups)}])
                 OR prj_owner = {user_id})
+                ORDER BY last_change_date DESC
                 """
             )
-
-            db_projects = await self.__load_projects(
-                conn, query, user_id, user_groups, filter_by_services
+            total_number_of_projects = await conn.scalar(
+                f"SELECT count(*) FROM ({query}) AS _pagination_query"
             )
 
-            projects_list.extend(db_projects)
+            prjs, prj_types = await self.__load_projects(
+                conn,
+                query + f"OFFSET {offset} LIMIT {limit or 'NULL'}",
+                user_id,
+                user_groups,
+                filter_by_services=filter_by_services,
+            )
 
-        return projects_list
+            return (
```

Comment:
so ``len(prjs) <= total_number_of_projects``? right? Speak with @odeimaiz . Perhaps at the end is better to transmit them all and "mark" the ones with services not accessibles.

Reply:
yes. I already spoke with @odeimaiz about this. but I want to do it in a separate PR.